Sphinx fails out when run at root directory inside of a container

### Bug Description

Attempting to local test container builds for sphinx documentation I ran across an issue where after completing QuickStart and attempting to run make html you get returned the following error. It appears that it fails to grab the intial `p` in `/proc` before attempting to navigate to `cwd` and due to this it does not find the file at all.

**Example error:**

```root@38d46495819d:/# make html
Running Sphinx v4.3.2
/roc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
^C
Interrupted!
make: *** [Makefile:20: html] Interrupt

root@38d46495819d:/#
```


**Example index.rst**

```root@38d46495819d:/# cat index.rst 
.. dsa documentation master file, created by
   sphinx-quickstart on Thu Jan  6 18:42:52 2022.
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

Welcome to dsa's documentation!
===============================

.. toctree::
   :maxdepth: 2
   :caption: Contents:



Indices and tables
==================

* :ref:`genindex`
* :ref:`modindex`
* :ref:`search`
root@38d46495819d:/# ```


### How to Reproduce

1. docker pull python:latest
2. podman run -it --entrypoint bash docker.io/python
3. pip3 install sphinx
4. sphinx-quickstart
5. fill in values recieve this output
```Creating file /conf.py.
Creating file /index.rst.
Creating file /Makefile.
Creating file /make.bat.

Finished: An initial directory structure has been created.

You should now populate your master file /index.rst and create other documentation
source files. Use the Makefile to build the docs, like so:
   make builder
where "builder" is one of the supported builders, e.g. html, latex or linkcheck.
```
6. run `make html`
```
root@38d46495819d:/# make html
Running Sphinx v4.3.2
making output directory... done
/roc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
/roc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/proc/1/cwd/index.rst: WARNING: document not readable. Ignored.
^C
Interrupted!
make: *** [Makefile:20: html] Interrupt

root@38d46495819d:/# 

```
### Expected behavior

**After running make html**

```
[tester@test containers]$ make html
Running Sphinx v4.3.0
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
reading sources... [100%] index                                                                                                                                                                                                                                        
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index                                                                                                                                                                                                                                         
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.
The HTML pages are in _build/html.
```


### Your project

N/A

### Screenshots

![image](https://user-images.githubusercontent.com/92595847/148435143-787fa871-7b48-41d8-a2cb-8f8781aa4ab8.png)

![image](https://user-images.githubusercontent.com/92595847/148435562-1aede0e3-cd08-4f9c-9e43-04ec8aa12548.png)


### OS

Debian GNU/Linux 11

### Python versions

3.10.1, 3.9.2, 3.8.1

### Sphinx version

4.3.2

### Sphinx extensions

_No response_

### Extra tools

_No response_

### Additional context

I would like to note that it generally it is not a good idea to normally install to root. Whether that be a warn or a documentation change.

--- a/CHANGES
+++ b/CHANGES
@@ -67,6 +67,8 @@ Bugs fixed
   at info-field-list
 * #9390: texinfo: Do not emit labels inside footnotes
 * #9979: Error level messages were displayed as warning messages
+* #10057: Failed to scan documents if the project is placed onto the root
+  directory
 
 Testing
 --------

--- a/sphinx/util/__init__.py
+++ b/sphinx/util/__init__.py
@@ -72,10 +72,11 @@ def get_matching_files(dirname: str,
     """
     # dirname is a normalized absolute path.
     dirname = path.normpath(path.abspath(dirname))
-    dirlen = len(dirname) + 1    # exclude final os.path.sep
 
     for root, dirs, files in os.walk(dirname, followlinks=True):
-        relativeroot = root[dirlen:]
+        relativeroot = path.relpath(root, dirname)
+        if relativeroot == ".":
+            relativeroot = ""  # suppress dirname for files on the target dir
 
         qdirs = enumerate(path_stabilize(path.join(relativeroot, dn))
                           for dn in dirs)  # type: Iterable[Tuple[int, str]]

